src/netlink_namespace.c

```diff
@@ -688,18 +693,24 @@ static int ir_config(const char *key, const char *value) {
   return (status);
 } /* int ir_config */
 
-static void close_fds(int* selfns, int* targetns) {
-  if ((*selfns >= 0) && (close(*selfns) != 0)) {
-    ERROR("Closing \"selfns\" file failed: %m");
-  }
+static void ir_init_error(int* selfns, int* targetns) {
   if ((*targetns >= 0) && (close(*targetns) != 0)) {
     ERROR("Closing \"targetns\" file failed: %m");
   }
-} /* fd_close */
+  if (*selfns >= 0) {
+    if (setns(*selfns, CLONE_NEWNET) != 0) {
+      ERROR("netlink_namespace plugin: ir_init_error: Getting back to original namespace failed: %m");
```

Comment:
As per man setns:

RETURN VALUE
       On success, setns() returns 0.  On failure, -1 is returned and errno is set to indicate the error.

Adding abort if setns() fails.